src/plugins/data/server/search/routes/search.ts

```diff
@@ -12,82 +12,97 @@ import { reportServerError } from '@kbn/kibana-utils-plugin/server';
 import { getRequestAbortedSignal } from '../../lib';
 import type { DataPluginRouter } from '../types';
 
-export function registerSearchRoute(router: DataPluginRouter): void {
-  router.post(
-    {
-      path: '/internal/search/{strategy}/{id?}',
-      validate: {
-        params: schema.object({
-          strategy: schema.string(),
-          id: schema.maybe(schema.string()),
-        }),
+export const SEARCH_API_BASE_URL = '/internal/search';
 
-        body: schema.object(
-          {
-            legacyHitsTotal: schema.maybe(schema.boolean()),
-            sessionId: schema.maybe(schema.string()),
-            isStored: schema.maybe(schema.boolean()),
-            isRestore: schema.maybe(schema.boolean()),
+export function registerSearchRoute(router: DataPluginRouter): void {
+  router.versioned
+    .post({
+      path: `${SEARCH_API_BASE_URL}/{strategy}/{id?}`,
+      access: 'internal',
+    })
+    .addVersion(
+      {
+        version: '1',
```

Comment:
Sorry, I don't think I completely understand, why we are hard coding version here and not getting it from the request sent.

Reply:
Here's some context: https://docs.elastic.dev/kibana-dev-docs/versioning-http-apis#use-the-versioned-router

Each route can have multiple versions registered on the server. For example, the server might have a `v1` and `v2` registered. The client sends the latest version it knows about, which may not be the latest version the server has registered. This is the main purpose of the versioned router (to preserve backwards compatibility when the server has updated but the client hasn't yet).

Reply:
Okay.. understood. Thank you very much for the link and pardon my ignorance.